Simplified/AxisService/NYPLRSACypher.swift

```diff
@@ -126,52 +128,95 @@ struct NYPLRSACypher: NYPLRSACryptographing {
-                                                        messageData as CFData,
-                                                        nil)
-      else {
-        NYPLRSACypher
-          .logFailedEncryptionError(NYPLRSACypher.nilEncryptedDataFailure)
+        .logFailedInitializationError(
+          "Failed to get exponent and or modulus for public key")
         return nil
     }
     
-    return (encryptedData as Data).base64EncodedString()
+    let modulus = mod.base64EncodedString()
+    guard modulus.count >= 342 else {
+      NYPLRSACypher
+        .logFailedInitializationError("Got incorrect modulus from public key")
+      return nil
+    }
+    
+    self.modulus = String(modulus.prefix(342)).replacingOccurrences(of: "/", with: "_")
+    self.exponent = exp.base64EncodedString()
   }
   
-  func decryptText(_ encryptedString: String) -> String? {
-    let _messageData = Data(base64Encoded: encryptedString)
+  static private func parsePublicSecKey(publicKey: SecKey) -> (mod: Data, exp: Data)? {
     
-    guard let messageData = _messageData else {
-      NYPLRSACypher
-        .logFailedDecryptionError(NYPLRSACypher.stringToDataConversionFailure)
+    guard let pubAttributes = SecKeyCopyAttributes(publicKey) as? [String: Any] else {
       return nil
     }
     
+    let keySize = pubAttributes[kSecAttrKeySizeInBits as String] as! Int
+    let pubData = pubAttributes[kSecValueData as String] as! Data
+    var modulus = pubData.subdata(in: 8..<(pubData.count - 5))
+    let exponent = pubData.subdata(in: (pubData.count - 3)..<pubData.count)
+    
+    if modulus.count > keySize / 8 { // --> 257 bytes
+      modulus.removeFirst(1)
+    }
+    
+    return (mod: modulus, exp: exponent)
+  }
+  
+  func decryptWithPKCS1_OAEP(_ data: Data) -> Data? {
     guard let decryptData = SecKeyCreateDecryptedData(privateKeySec,
                                                       .rsaEncryptionOAEPSHA1,
-                                                      messageData as CFData,
+                                                      data as CFData,
                                                       nil)
       else {
         NYPLRSACypher
           .logFailedDecryptionError(NYPLRSACypher.nilDecryptedDataFailure)
         return nil
     }
+    return decryptData as Data
+  }
+  
+  func decryptWithAES(_ data: Data, key: Data) -> Data? {
+    let content = data.subdata(in: 4..<data.count)
+    let keyLength = key.count
+    let validKeyLengths = [kCCKeySizeAES128, kCCKeySizeAES192, kCCKeySizeAES256]
+    if (validKeyLengths.contains(keyLength) == false) {
+        return nil
+    }
+
+    let ivSize = kCCBlockSizeAES128;
+    let clearLength = size_t(content.count - ivSize)
+    var clearData = Data(count:clearLength)
+
+    var numBytesDecrypted :size_t = 0
+    let options   = CCOptions(kCCOptionPKCS7Padding)
+
+    let cryptStatus = clearData.withUnsafeMutableBytes {cryptBytes in
```

Comment:
if you have reference for this process, it would be great to add it to the code -- I'm sure this works, but I have no idea why.

Reply:
done